php7 + apache2 fix

commercebyte opened the pull request.

Commit:
Fix seg fault in apache2 + php7 on shutdown,
check if store is null

Commit:
The issue with PHP7 + Apache2 seg fault is properly addressed and tested.
Explicitly restoring the dtor handlers on shutdown

colder committed:
Fix build warnings

colder commented:
I just pushed a fix for a build warning to topic/fix-segfault. Can you confirm that this new version works as expected on your end? As soon as you do I will merge and release.

https://github.com/colder/php-weakref/tree/topic/fix-segfault

commercebyte commented:
Installed, looks like it works fine, thanks for cleaning up!
______________
Jim Zubov
CommerceByte Consulting
President

On March 15, 2016 6:11:39 AM EDT, Etienne Kneuss <notifications@github.com> wrote:
>I just pushed a fix for a build warning to topic/fix-segfault. Can you
>confirm that this new version works as expected on your end? As soon as
>you do I will merge and release.
>
>https://github.com/colder/php-weakref/tree/topic/fix-segfault
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/colder/php-weakref/pull/25#issuecomment-196753361

Commit:
wr_store.c:
old_dtors it actually holds zend_object_dtor_obj_t's, not zval's, so Z_PTR_P() didn't do any good

commercebyte commented:
I found a problem, pushed a fix, see comment on commit -
______________
Jim Zubov
CommerceByte Consulting
President

On March 15, 2016 6:11:39 AM EDT, Etienne Kneuss <notifications@github.com> wrote:
>I just pushed a fix for a build warning to topic/fix-segfault. Can you
>confirm that this new version works as expected on your end? As soon as
>you do I will merge and release.
>
>https://github.com/colder/php-weakref/tree/topic/fix-segfault
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/colder/php-weakref/pull/25#issuecomment-196753361
